sku.0/sys.server/compiled/game/script/test/qatool.java

```diff
@@ -4182,7 +4190,27 @@ public boolean createEnzyme(obj_id self, StringTokenizer st) throws InterruptedE
         }
         return true;
     }
-    public boolean createLyase(obj_id self) throws InterruptedException
+    /**
+     * Retrieves an enzyme color from the next token in a string tokenizer.
+     * Will return a random color index if one can not be determined.
+     */
+    private int getEnzymeColorFromNextToken(obj_id self, StringTokenizer st) {
+        if(st.hasMoreTokens()) {
+            // Lowercase and remove underscores so that colors such as Dark_Blue will still work.
+            var colorName = st.nextToken().toLowerCase().replace("_", "");
+
+            var colorIndex = ENZYME_COMMAND_COLOR_LOOKUP.indexOf(colorName);
+
+            if(colorIndex >= 0) {
+                return colorIndex;
+            }
+
+            sendSystemMessageTestingOnly(self, "Invalid color specified, random color will be used. Valid colors are: " +  String.join(", ", ENZYME_COMMAND_COLOR_LOOKUP));

```

Comment:
This doesn't really matter but for the sake of consistency with how most user-input related admin commands are handled: If the user's input isn't a valid option for the parameter, it is probably better to provide the system message correcting the mistake but return so as not to just create an enzyme for the hell of it when they typed in an incorrect color but obviously intended to get a specific color.

if color is null/equalsIgnoreCase("") -> return random
if color is in list -> return color
else, return script continue with correction system message

Reply:
Yep - that's totally fair! I've pushed up a commit to do that now.